demo/components/auto-complete.demo.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import {AutoComplete} from '../../src';
+import {ChangeEvent} from '../../src/types/events';
+
+export interface AutoCompleteDemoState {
+    open: boolean;
+    inputText: string;
+}
+
+const items = ['Muffins', 'Pancakes', 'Cupcakes', 'Souffles',
+               'Pasta', 'Soup', 'Caramel', 'Avazim', 'Moses'];
+
+export class AutoCompleteDemo extends React.Component<{}, AutoCompleteDemoState> {
+    public state = {open: false, inputText: ''};
+
+    public onChange = (e: ChangeEvent<string>) => {
+        this.setState({
+            inputText: e.value
+        });
+    }
+
+    public toggleOpen = (e: ChangeEvent<boolean>) => {
```

Comment:
This means we have implemented incorrectly controlled component. This event should just set the "open" state to `e.value`. The demo shouldn't do actual toggling! It should just pass the state. Something's wrong in the component implementation. We have btw the same mistake in the dropdown demo.